apis/python/src/tiledbsoma/io/spatial/outgest.py

```diff
@@ -406,3 +408,215 @@ def to_spatial_data_multiscale_image(
     )
 
     return images_to_datatree(image_data_arrays)
+
+
+def _get_transform_from_collection(
+    key: str, metadata: Mapping[str, Any]
+) -> Optional[somacore.CoordinateTransform]:
+    transform_key = f"soma_scene_registry_{key}"
+    if transform_key in metadata:
+        transform_json = metadata[transform_key]
+        return transform_from_json(transform_json)
+    return None
+
+
+def _add_scene_to_spatial_data(
+    sdata: sd.SpatialData,
+    scene_id: str,
+    scene: Scene,
+    *,
+    obs_id_name: str,
+    var_id_name: str,
+    measurement_names: Optional[Sequence[str]] = None,
+) -> None:
+    """Adds items from a Scene to a SpatialData object.
+
+    Args:
+        sdata: SpatialData object to update.
+        scene_id: Name of the scene that will be added to the SpatialData object.
+        measurement_names: Measur
+
+    """
+    # Cannot have spatial data if no coordinate space.
+    if scene.coordinate_space is None:
+        return
+
+    # Get the map from Scene dimension names to SpatialData dimension names.
+    input_axis_names = scene.coordinate_space.axis_names
+    _, scene_dim_map = _convert_axis_names(input_axis_names, input_axis_names)
+
+    # Export obsl data to SpatialData.
+    if "obsl" in scene:
+        for key, df in scene.obsl.items():
+            output_key = f"{scene_id}_{key}"
+            transform = _get_transform_from_collection(key, scene.obsl.metadata)
+            if isinstance(df, PointCloudDataFrame):
+                if "soma_geometry" in df.metadata:
+                    sdata.shapes[output_key] = to_spatial_data_shapes(
```

Comment:
There are opportunities here (FYI for later/someday) to use thread-pooling to parallelize these for-loops. Bruce did similar for non-spatial I/O recently.